### PR TITLE
Use `SCRIPT_NAME` instead of route prefix & configure gunicorn for secure scheme header

### PR DESCRIPTION
**Pull request by ltshb**
The path prefix is now set by the deployment using the WSGI SCRIPT_NAME
environment variable. The change in this paradigma come from this post: https://dlukes.github.io/flask-wsgi-url-prefix.html

Also configure gunicorn to use secure scheme header which allow us to
get rid of the middleware. This settings is required in order to have the correct proto in generated urls in responses.

Change the default ALLOWED_ORIGIN to allow everything.

### CODE REVIEW

**Review by hansmannj — APPROVED** [review]
Thanks, looks good to me :+1:

**Review by rebert — APPROVED** [review]
@ltshb thanks for the investigation

File: Makefile [hunk 1]
```diff
@@ -43,6 +43,7 @@ PIP_FILE_LOCK = Pipfile.lock
 # default configuration
 ENV_FILE ?= .env.default
 HTTP_PORT ?= 5000
+ROUTE_PREFIX ?= /api/kml
```
Comment (rebert) [inline Makefile:46]: WHY not using SCRIPT_NAME allover?

Comment (ltshb) [reply Makefile:46]: What do you mean exactly ? We cannot use the SCRIPT_NAME with the Flask development server, only with gunicorn.

File: README.md [hunk 2]
```diff
@@ -28,11 +28,11 @@ A detailed descriptions of the endpoints can be found in the [OpenAPI Spec](http
...
 
-| Environments | URL                                                                                                                   |
-| ------------ | --------------------------------------------------------------------------------------------------------------------- |
-| DEV          | [https://service-kml.bgdi-dev.swisstopo.cloud/api/kml/](https://service-kml.bgdi-dev.swisstopo.cloud/api/kml/)  |
-| INT          | [https://service-kml.bgdi-int.swisstopo.cloud/api/kml/](https://service-kml.bgdi-int.swisstopo.cloud/api/kml/)  |
-| PROD         | [https://service-kml.bgdi-prod.swisstopo.cloud/api/kml/](https://service-kml.bgdi-int.swisstopo.cloud/api/kml/) |
+| Environments | URL                                                                                 |
+| ------------ | ----------------------------------------------------------------------------------- |
+| DEV          | [https://sys-publid.dev.bgdi.ch/api/kml/](https://sys-publid.dev.bgdi.ch/api/kml/)  |
+| INT          | [https://sys-publid.int.bgdi.ch/api/kml/](https://sys-publid.int.bgdi.ch/api/kml/)  |
```
Comment (IsabelleBzr) [inline README.md:34]: Small typing error, public

**Review by IsabelleBzr — APPROVED** [review]
:+1:  Thanks

**Pull request merged**